RATIONALE.md

```diff
@@ -1,24 +1,34 @@
 # zipkin-docker-jre rationale
 
-## Why do we link /bin/sh to BusyBox?
+## Why do we use Alpine Linux instead of Distroless?
 
-It is possible to override some Docker commands to use something besides
-`/bin/sh`, notably to allow use of `/busybox/sh` included in Alpine and
-Distroless debug images:
+### We need a working /bin/sh
 
-Ex.
-```diff
--ENTRYPOINT ["/busybox/sh", "run.sh"]
-+ENTRYPOINT run.sh
+Alpine Linux includes `/busybox/sh`, linked to `/bin/sh`. This allows us to
+customize startup and add HEALTHCHECK instructions. Notably, Alpine linking
+`/bin/sh` allows docker-compose to override health checks as necessary. This is
+because docker-compose always invokes its health check overrides with
+`/bin/sh -c`.
+
+It is possible to layer BusyBox on a distroless image to perform the same, but
+Alpine does this by default.
+
+### We want the smallest base image
+
+We are often criticized for the size of our Docker image, as it is an ancillary
+part of people's environments. Zipkin's 'slim' build is currently a 26MB jar
+file. The bulk of the size left is the JRE. Here's a comparison of size between
+a comparible build of the same JRE, one with Distroless (Debian based) and the
+other Alpine.
+
+```
+openzipkin/jre-full                  alpine              128317d6038e        20 seconds ago      87.1MB
+openzipkin/jre-full                  distroless          0717ad881158        3 minutes ago       102MB
 ```
 
-However, there are some inconsistencies. For example, it is possible to override
-docker-compose health-check to use a different shell, but not in Dockerfile
-syntax (always invokes with `/bin/sh -c`).
+### We are ok not having BoringSSL in the default image
```

Comment:
TODO: try instructions here for boringssl (and find a way to validate them :P) https://github.com/grpc/grpc-java/blob/master/SECURITY.md#netty. If not huge, consider adding to the image

@anuraaga @trustin I suppose we can just look at armeria startup for verification of BoringSSL working? I thought I remember a log statement.. any other quick ideas welcome.

Reply:
prefer shell commands for verification as I can built it into a check here..

Reply:
I think by default it logs whether it's able to load boring or not, and epoll too. But if it didn't, dumpOpenSslInfo should definitely do the trick

https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/Flags.java#L538

Reply:
this adds only 600K and we are net down anyway. I'll speculatively add it to the image